deepmd/pt/model/task/polarizability.py

```diff
@@ -184,7 +190,62 @@ def compute_output_stats(
             The path to the stat file.
 
         """
-        pass
+        if self.shift_diag:
+            if stat_file_path is not None:
+                stat_file_path = stat_file_path / "constant_matrix"
+            if stat_file_path is not None and stat_file_path.is_file():
+                constant_matrix = stat_file_path.load_numpy()
+            else:
+                if callable(merged):
+                    # only get data for once
+                    sampled = merged()
+                else:
+                    sampled = merged
+
+                sys_constant_matrix = []
+                for sys in range(len(sampled)):
+                    nframs = sampled[sys]["type"].shape[0]
+
+                    if sampled[sys]["find_atomic_polarizability"] > 0.0:
+                        sys_atom_polar = compute_stats_from_atomic(
+                            sampled[sys]["atomic_polarizability"].numpy(force=True),
+                            sampled[sys]["type"].numpy(force=True),
+                        )[0]
+                    else:
+                        if not sampled[sys]["find_polarizability"] > 0.0:
+                            continue
+                        sys_type_count = np.zeros((nframs, self.ntypes))
+                        for itype in range(self.ntypes):
+                            type_mask = sampled[sys]["type"] == itype
+                            sys_type_count[:, itype] = type_mask.sum(dim=1).numpy(
+                                force=True
+                            )
+
+                        sys_bias_redu = sampled[sys]["polarizability"].numpy(force=True)
+
+                        sys_atom_polar = compute_stats_from_redu(
+                            sys_bias_redu, sys_type_count
+                        )[0]
+                    cur_constant_matrix = np.zeros(self.ntypes)
+                    if sys_atom_polar.shape[0] < self.ntypes:
```

Comment:
it seems that the case should not happen. The dataloader is initialized by the model's type_map, so it is expected to return the data containing the same number of types as the model. @iProzd @CaRoLZhangxy please check. 
see also 
https://github.com/deepmodeling/deepmd-kit/blob/fa8e6453b5cc88ae575284c4c0ded06758c6de82/deepmd/pt/utils/dataloader.py#L60-L67